library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -732,7 +792,34 @@ private void checkVisitedDirectory(Path path)
    */
   protected void scanDirectory()
   {
-    if (System.currentTimeMillis() - scanIntervalMillis >= lastScanMillis) {
+    boolean allowScan = true;
+    if (emitBatchTuples) {
+      if (canEmitEndBatch) {
+        /*
+         * Emit EndBatch and mark canEmitEndBatch to false to avoid multiple emits of End Batch
+         */
+        emitEndBatchControlTuple();
+        endBatchEmittedInCurrentWindow = true;
+        currentWindowState.endBatchEmitted = true;
+        canEmitEndBatch = false;
+      }
```

Comment:
A scan is tried again only after all files are processed. Also, canEmitEndBatch is true only if start batch is emitted earlier. 
Need to add a "return" statement here to ensure startBatch also does not get emitted in same window.